docs/guides/src/main/server/caching.adoc

```diff
@@ -0,0 +1,95 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+title="Configure distributed caches"
+summary="Understand how to configure the caching layer"
+includedOptions="cache cache-*">
+
+Keycloak is designed for high availability and multi-node clustered setups. The current distributed cache implementation is built on top of https://infinispan.org[Infinispan], a high-performance, distributable in-memory data grid.
+
+All available cache options are build options, so they need to be applied to a `build` of Keycloak before starting.
+
+== Enable distributed caching
+Caching is enabled per default when you start Keycloak in production mode, using the `start` command. This will automatically discover other Keycloak nodes in your network. To explicitly enable distributed infinispan caching, use:
+
+<@kc.build parameters="--cache=ispn"/>
+
+When you start Keycloak in development mode, using the `start-dev` command, Keycloak defaults to only use local caches using `--cache=local` instead. Using the `local` cache mode is intended only for development and testing purposes.
+
+== Configure Caches
+Keycloak provides a cache configuration file with sensible defaults located at `conf/cache-ispn.xml`.
+
+The cache configuration is a regular https://infinispan.org/docs/stable/titles/configuring/configuring.html[Infinispan] configuration file.
+
+=== Overview: Cache types and Defaults
+The following table gives an overview of the specific caches Keycloak uses. These caches are configured in `conf/cache-ispn.xml`:
+
+|====
+|Cache name|Cache Type|Description
+|realms|Local|Cache persisted realm data
+|users|Local|Cache persisted user data
+|authorization|Local|Cache persisted authorization data
```

Comment:
Would mention that these are invalidated through events in the `work` cache

Reply:
Would it suffice to add a sentence around that in the paragraph below? Alternatively, I could split this up into two tables, one for the local ones with paragraph below, one for the distributed/replicated ones. wdyt?